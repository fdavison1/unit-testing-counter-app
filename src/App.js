import React from 'react'
import './App.css'

export default class App extends React.Component {
  constructor(){
    super()
    this.state = {
      count: 0
    }
  }
  makeIncrementer = amount => () => 
  this.setState(prevState => ({
    count: prevState.count + amount
  }))

  increment = this.makeIncrementer(1)

  decrement = this.makeIncrementer(-1)

  render(){
    return(
      <div className='App'>
        <p>Count: {this.state.count}</p>
        
        <button 
        className='increment'
        onClick={this.increment}>
        Increment
        </button>

        <button
          className='decrement'
          onClick={this.decrement}
        >Decrement
        </button>

      </div>
    )
  }
}