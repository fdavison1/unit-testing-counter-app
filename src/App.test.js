import React from 'react'
import { shallow } from 'enzyme'
import App from './App'
import './setupTests'


describe('App Component', () => {
  it('starts with a count of 0', () => {
    const wrapper = shallow(<App />)
    const text = wrapper.find('p').text()
    expect(text).toEqual('Count: 0')
  })
})