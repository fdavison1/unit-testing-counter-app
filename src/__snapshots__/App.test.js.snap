// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component matches the snapshot 1`] = `
<div
  className="App"
>
  <p>
    Count: 
    0
  </p>
  <button
    className="increment"
    onClick={[Function]}
  >
    Increment
  </button>
  <button
    className="decrement"
    onClick={[Function]}
  >
    Decrement
  </button>
</div>
`;
